cpp/oneapi/dal/algo/dbscan/test/batch.cpp

```diff
@@ -100,10 +105,68 @@ class dbscan_batch_test : public te::float_algo_fixture<std::tuple_element_t<0,
         CAPTURE(val, ref_val, fabs(val - ref_val) / max_abs, ref_tol);
         return fabs(val - ref_val) / max_abs < ref_tol;
     }
+
+    void mode_checks(result_option_id compute_mode,
+                     const table& data,
+                     const table& weights,
+                     Float epsilon,
+                     std::int64_t min_observations) {
+        CAPTURE(epsilon, min_observations);
+
+        INFO("create descriptor")
+        const auto dbscan_desc =
+            get_descriptor(epsilon, min_observations).set_result_options(compute_mode);
+
+        INFO("run compute");
+        const auto compute_result =
+            oneapi::dal::test::engine::compute(this->get_policy(), dbscan_desc, data, weights);
+
+        INFO("check mode");
+        check_for_exception_for_non_requested_results(compute_mode, compute_result);
+    }
+
+    void check_for_exception_for_non_requested_results(result_option_id compute_mode,
+                                                       const result_t& result) {
+        if (!compute_mode.test(result_options::responses)) {
+            REQUIRE_THROWS_AS(result.get_responses(), domain_error);
+        }
+        if (!compute_mode.test(result_options::core_flags)) {
+            REQUIRE_THROWS_AS(result.get_core_flags(), domain_error);
+        }
+        if (!compute_mode.test(result_options::core_observations)) {
+            REQUIRE_THROWS_AS(result.get_core_observations(), domain_error);
+        }
+        if (!compute_mode.test(result_options::core_observation_indices)) {
+            REQUIRE_THROWS_AS(result.get_core_observation_indices(), domain_error);
+        }
+    }
 };
 
 using dbscan_types = COMBINE_TYPES((float, double), (dbscan::method::brute_force));
 
+TEMPLATE_LIST_TEST_M(dbscan_batch_test,
+                     "dbscan compute mode check",
+                     "[dbscan][batch]",
+                     dbscan_types) {
+    using Float = std::tuple_element_t<0, TestType>;
```

Comment:
Please use `float_t` instead. We have an agreement that only type in template parameters should be called in Camel Case. In all other cases it should be done in snake case like `float_t`

Reply:
Done